package/MDAnalysis/analysis/distances.py

```diff
@@ -202,3 +204,65 @@ def between(group, A, B, distance):
     resA = ns_group.search(A, distance)
     resB = ns_group.search(B, distance)
     return resB.intersection(resA)
+
+
+class AtomicDistances(AnalysisBase):
+    r"""Class to calculate atomic distances between two AtomGroups over a
+    trajectory.
+
+    Parameters
+    ----------
+    ag1, ag2 : AtomGroup
+        :class:`~MDAnalysis.core.groups.AtomGroup` with the
+        same number of atoms
+    pbc : bool, optional
+        If ``True``, calculates atomic distances with periodic boundary
+        conditions (PBCs). The unitcell dimensions of the system must be
+        orthogonal or triclinic for the calculations with PBCs to work.
+        Defaults to ``True``.
+
+    Attributes
+    ----------
+    results : :class:`numpy.ndarray`
+        The distances `|ag1[i] - ag2[i]|` for all `i` from `0` to
+        `n_atoms - 1` for each frame over the trajectory.
+    n_frames : int
+        Number of frames included in the analysis.
+    n_atoms : int
+        Number of atoms in each atom group.
```

Comment:
I don't see where these variables are set?

Reply:
Thanks for all the feedback! I learned a lot working through it, and I will push another commit after I sort out the empty Universe question I have (I am not sure how to make an empty Universe with multiple frames and I asked in the Discord)

I added those variables because other classes seemed to have them and I used those as reference. 

`results` refers to `self.results` in `def _prepare(self)` and `def _single_frame(self)`
`n_frames` is used to make the NumPy array in `def _prepare(self)`
`n_atoms` is used to check that `ag1` and `ag2` have the same number of atoms and in `def _prepare(self)` to make the NumPy array

Would it be better to remove some/all of them?